ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala

```diff
@@ -238,33 +239,58 @@ object JsonProtocol extends DefaultJsonProtocol with ExtraFormats {
     *  1. template IDs are required
     *  2. query key may be absent
     *  3. special error if you appear to have queried outside 'query'
+    *
+    *  This provides an (almost) consistent behavior when reading the 'templateIds' and
+    *  'query' fields. Further extra fields may be added by concrete implementations.
     */
-  implicit val GetActiveContractsRequestFormat: RootJsonReader[domain.GetActiveContractsRequest] = {
-    case class GACR(
+  private def requestJsonReader[Request](validExtraFields: Set[String])(
```

Comment:
probably not for this PR but this seems like something that should have unit tests.